spring-cloud-gcp-samples/spring-cloud-gcp-pubsub-sample/src/main/java/com/example/WebController.java

```diff
@@ -84,17 +84,20 @@ public RedirectView publish(@RequestParam("topicName") String topicName,
 
 	@GetMapping("/pull")
 	public RedirectView pull(@RequestParam("subscription1") String subscriptionName) {
-
+		//tag::pull10[]
```

Comment:
I don't think it is a good idea to pollute samples code with some metadata.
This code is going to be used by end-users and they are going to ask questions what is this about and how to deal with it.

Reply:
@dmitry-s Any way we can make the comment actually useful and still work for embedding in docs?

Reply:
We can add any text before the tag name (tag::pull10[])

Reply:
I am also not quite a fan of using the asciidoc metadata. It seems like an error-prone process; even in your current draft it looks like [some of the code snippets did not render correctly](https://github.com/spring-cloud/spring-cloud-gcp/blob/8ccfac1a07771f8e3510f0a54f4f01b077876f9c/docs/src/main/asciidoc/pubsub.adoc). Once we get to the point where we have many snippets, it is going to get confusing to have many references between code and docs.

I would prefer if we just copy/pasted code snippets in the doc. Yes there is a risk they may go out of sync and break, but this risk would be reduced significantly if we used snippets sparingly and kept them small. For example, it is not necessary to have snippets for all Pub/Sub actions like `createTopic`, `deleteTopic`, etc. since these are self-explanatory. It's probably only necessary to have a few code snippets for `Beans` to include in autoconfig and maybe complicated things like how to convert PubsubMessage to String. Such snippets would be small enough that they would be robust to breakage.

Reply:
Well, why then just not link to the existing samples which always are going to be up-to-date.

In other our projects we really have burden to support snippets in the docs as is without any includes and links.
Right, we sometimes forget to update them over there, but in the end it is less work, then answering for questions in issues or StackOverflow.

Reply:
We do link the samples, but it turns out users prefer seeing code snippets in the doc (based on feedback).

I think I'll just create a separate test class for ref doc code snippets, from which they will be included. This way samples can stay clean of metadata.

Reply:
Gotcha, I like this idea. Is your intention to include Java code in the current `docs/`module to reference? I think this would be a good idea; basically keeping the code snippets and refdoc in the same place.